<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>merFISH Mouse Hypothalmic Preoptic Region â€¢ Giotto Suite</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="merFISH Mouse Hypothalmic Preoptic Region">
<meta property="og:description" content="Giotto">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-BCXT5392QC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BCXT5392QC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0.997</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Start
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/getting_started.html#installation">Installation</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#getting-started">Getting Started</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#analyses">Analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datasets_overview.html#array-based-datasets-1">Array based</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#single-cell-1">Single-cell</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#subcellular-datasets-1">Subcellular</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#protein-multiplexing-1">Protein multiplexing</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/contribute.html">Contribute</a>
</li>
<li>
  <a href="../articles/giotto_news.html">News</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faqs.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">GitHub Issues Guidelines</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Try Giotto
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Docker</li>
    <li class="dropdown-header">Binder</li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RubD/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>merFISH Mouse Hypothalmic Preoptic Region</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RubD/Giotto/blob/HEAD/vignettes/merFISH_hypoth_220915.Rmd" class="external-link"><code>vignettes/merFISH_hypoth_220915.Rmd</code></a></small>
      <div class="hidden name"><code>merFISH_hypoth_220915.Rmd</code></div>

    </div>

    
    
<p>Updated <code>9/15/2022</code>. <em>Giotto</em> version
<code>2.0.0.998</code>.</p>
<p>Please check the version you are using to get the same results.</p>
<p>Clustering, 3D visualization, and cell type identification of
clusters using marker genes will be explored in this tutorial.</p>
<div class="section level2">
<h2 id="start-giotto">Start Giotto<a class="anchor" aria-label="anchor" href="#start-giotto"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Specify path from which data may be retrieved/stored</span></span>
<span><span class="va">data_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_visual_data/'</span><span class="op">)</span></span>
<span><span class="co"># alternatively, "/path/to/where/the/data/lives/"</span></span>
<span></span>
<span><span class="co"># Specify path to which results may be saved</span></span>
<span><span class="va">results_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_visual_results/'</span><span class="op">)</span> </span>
<span><span class="co"># alternatively, "/path/to/store/the/results/"</span></span>
<span></span>
<span><span class="co"># Ensure Giotto environment lives local to the machine</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu"><a href="../reference/checkGiottoEnvironment.html">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu"><a href="../reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dataset-explanation">Dataset Explanation<a class="anchor" aria-label="anchor" href="#dataset-explanation"></a>
</h2>
<p><a href="https://doi.org/10.1126/science.aau5324" class="external-link">Moffitt et al.</a>
created a 3D spatial expression dataset consisting of 155 genes from ~1
million single cells acquired from mouse hypothalamic preoptic regions.
Please ensure that <a href="https://www.gnu.org/software/wget/?" class="external-link">wget</a> is installed locally
to streamline the download.</p>
</div>
<div class="section level2">
<h2 id="dataset-download">Dataset Download<a class="anchor" aria-label="anchor" href="#dataset-download"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># In the event of authentication issues with wget, </span></span>
<span><span class="co"># add ", extra = '--no-check-certificate' " after the method argument.</span></span>
<span><span class="co"># Get the dataset:</span></span>
<span><span class="fu"><a href="../reference/getSpatialDataset.html">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">'merfish_preoptic'</span>, directory <span class="op">=</span> <span class="va">data_directory</span>, method <span class="op">=</span> <span class="st">'wget'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="part-1-create-giotto-instructions-prepare-data">Part 1: Create Giotto Instructions &amp; Prepare Data<a class="anchor" aria-label="anchor" href="#part-1-create-giotto-instructions-prepare-data"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Optional, but encouraged: Set Giotto instructions</span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>save_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  save_dir <span class="op">=</span> <span class="va">results_directory</span>, </span>
<span>                                  python_path <span class="op">=</span> <span class="va">my_python_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create file paths to feed data into Giotto Object</span></span>
<span><span class="va">expr_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_data_expression.txt.gz"</span><span class="op">)</span></span>
<span><span class="va">loc_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_data_cell_locations.txt"</span><span class="op">)</span></span>
<span><span class="va">meta_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_metadata.txt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="part-2-create-giotto-object-process-data">Part 2: Create Giotto Object &amp; Process Data<a class="anchor" aria-label="anchor" href="#part-2-create-giotto-object-process-data"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create Giotto object</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expr_path</span>,</span>
<span>                                      spatial_locs <span class="op">=</span> <span class="va">loc_path</span>,</span>
<span>                                      instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Add additional metadata</span></span>
<span><span class="va">metadata</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">meta_path</span><span class="op">)</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellMetadata.html">addCellMetadata</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                                  new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">layer_ID</span>, </span>
<span>                                  vector_name <span class="op">=</span> <span class="st">'layer_ID'</span><span class="op">)</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellMetadata.html">addCellMetadata</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>                                  new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">orig_cell_types</span>,</span>
<span>                                  vector_name <span class="op">=</span> <span class="st">'orig_cell_types'</span><span class="op">)</span></span></code></pre></div>
<p>First pre-test filter parameters for both features and cells.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, detection <span class="op">=</span> <span class="st">'feats'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/0-filterDistributions.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterDistributions.html">filterDistributions</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, detection <span class="op">=</span> <span class="st">'cells'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/1-filterDistributions.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterCombinations.html">filterCombinations</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>                  expression_thresholds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1e-6</span>,<span class="fl">1e-5</span><span class="op">)</span>,</span>
<span>                  feat_det_in_min_cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500</span>, <span class="fl">1000</span>, <span class="fl">1500</span><span class="op">)</span>,</span>
<span>                  min_det_feats_per_cell <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/2-filterCombinations.png" style="width:50.0%"></p>
<p>Use the previously generated plots to inform filter decisions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Data Processing</span></span>
<span></span>
<span><span class="co"># Filter data</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterGiotto.html">filterGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>,</span>
<span>                                feat_det_in_min_cells <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                                min_det_feats_per_cell <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co"># Normalize data</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalizeGiotto.html">normalizeGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                   scalefactor <span class="op">=</span> <span class="fl">10000</span>, </span>
<span>                                   verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add statistics to Giotto Object</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/addStatistics.html">addStatistics</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                 expression_values <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Adjust for covariates</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/adjustGiottoMatrix.html">adjustGiottoMatrix</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, expression_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>                                      batch_columns <span class="op">=</span> <span class="cn">NULL</span>, covariate_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'layer_ID'</span><span class="op">)</span>,</span>
<span>                                      return_gobject <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                      update_slot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'custom'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now, take a glance at the data in both 2D and 3D.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 2D</span></span>
<span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/3-spatPlot2D.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="co"># 3D</span></span>
<span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.25</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/4-spat3D.png"></p>
</div>
<div class="section level2">
<h2 id="part-3-dimension-reduction">Part 3: Dimension Reduction<a class="anchor" aria-label="anchor" href="#part-3-dimension-reduction"></a>
</h2>
<p>There are only 155 genes within this dataset. Use them all (default)
within the dimension reduction.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                          feats_to_use <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                          scale_unit <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                          center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View details about the principal components</span></span>
<span><span class="fu"><a href="../reference/screePlot.html">screePlot</a></span><span class="op">(</span><span class="va">merFISH_gobject</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/5-screePlot.png" style="width:50.0%"></p>
<p>After the PCA, a UMAP may be run. Run the UMAP so clusters may be
visualized upon it.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                           dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, </span>
<span>                           n_components <span class="op">=</span> <span class="fl">3</span>, </span>
<span>                           n_threads <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> </span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/6-UMAP_3D.png"></p>
</div>
<div class="section level2">
<h2 id="part-4-cluster">Part 4: Cluster<a class="anchor" aria-label="anchor" href="#part-4-cluster"></a>
</h2>
<p>Create a nearest network, then perform Leiden clustering. The
clusters may be visualized on a UMAP.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a sNN network (default)</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                        dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, </span>
<span>                                        k <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span><span class="co"># Leiden cluster</span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                   resolution <span class="op">=</span> <span class="fl">0.2</span>, </span>
<span>                                   n_iterations <span class="op">=</span> <span class="fl">200</span>,</span>
<span>                                   name <span class="op">=</span> <span class="st">'leiden_0.2_200'</span><span class="op">)</span></span>
<span><span class="co"># Plot the clusters upon the UMAP</span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'leiden_0.2_200'</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            show_center_label <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/7-UMAP_3D.png"></p>
</div>
<div class="section level2">
<h2 id="part-5-co-visualize">Part 5: Co-Visualize<a class="anchor" aria-label="anchor" href="#part-5-co-visualize"></a>
</h2>
<p>View the clusters in-tissue on each layer.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'leiden_0.2_200'</span>, </span>
<span>            group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, </span>
<span>            cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">260</span>, <span class="fl">160</span>, <span class="fl">60</span>, <span class="op">-</span><span class="fl">40</span>, <span class="op">-</span><span class="fl">140</span>, <span class="op">-</span><span class="fl">240</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/8-spatPlot2D.png" style="width:50.0%"></p>
</div>
<div class="section level2">
<h2 id="part-6-cell-type-marker-gene-detection">Part 6: Cell Type Marker Gene Detection<a class="anchor" aria-label="anchor" href="#part-6-cell-type-marker-gene-detection"></a>
</h2>
<p>Call <strong>findMarkers_one_vs_all</strong> to identify marker
features. Click the function to see alternate methods, or look <a href="../docs/reference/findGiniMarkers.html">here</a> for details on
the gini method. Once marker features have been determined, observe the
differential expression across clusters within the violin plot.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>,</span>
<span>                                 method <span class="op">=</span> <span class="st">'gini'</span>,</span>
<span>                                 expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">'leiden_0.2_200'</span>,</span>
<span>                                 min_feats <span class="op">=</span> <span class="fl">1</span>, rank_score <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># Display details about the marker genes in-console</span></span>
<span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># violinplot</span></span>
<span><span class="va">topgini_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, feats <span class="op">=</span> <span class="va">topgini_genes</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_0.2_200'</span>, strip_position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/9-violinPlot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">topgini_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">6</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_0.2_200'</span><span class="op">)</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">topgini_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/10-plotMetaDataHeatmap.png" style="width:50.0%"></p>
</div>
<div class="section level2">
<h2 id="part-7-cell-type-annotation">Part 7: Cell-Type Annotation<a class="anchor" aria-label="anchor" href="#part-7-cell-type-annotation"></a>
</h2>
<p>Use known marker and differentially expressed genes (DEGs) to
identify cell type for each cluster.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Known markers and DEGs</span></span>
<span><span class="va">selected_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Myh11'</span>, <span class="st">'Klf4'</span>, <span class="st">'Fn1'</span>, <span class="st">'Cd24a'</span>, <span class="st">'Cyr61'</span>, <span class="st">'Nnat'</span>, <span class="st">'Trh'</span>, </span>
<span>                   <span class="st">'Selplg'</span>, <span class="st">'Pou3f2'</span>, <span class="st">'Aqp4'</span>, <span class="st">'Traf4'</span>, <span class="st">'Pdgfra'</span>, <span class="st">'Opalin'</span>, </span>
<span>                   <span class="st">'Mbp'</span>, <span class="st">'Ttyh2'</span>, <span class="st">'Fezf1'</span>, <span class="st">'Cbln1'</span>, <span class="st">'Slc17a6'</span>, <span class="st">'Scg2'</span>, <span class="st">'Isl1'</span>, <span class="st">'Gad1'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gobject_cell_metadata</span> <span class="op">=</span> <span class="fu"><a href="../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">merFISH_gobject</span><span class="op">)</span></span>
<span><span class="va">cluster_order</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gobject_cell_metadata</span><span class="op">$</span><span class="va">leiden_0.2_200</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_0.2_200'</span><span class="op">)</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">selected_genes</span>,</span>
<span>                    custom_feat_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/flip.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">selected_genes</span><span class="op">)</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="va">cluster_order</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/11-plotMetaDataHeatmap.png" style="width:50.0%"></p>
<p>Since there are more Leiden clusters than there are types of cells in
this dataset, the same cell type may be assigned to different cluster
numbers. This may be done only after verifying that particular clusters
highly express marker genes corresponding to the same cell type. The
above heatmap is used to streamline this process. Call
<strong>annotateGiotto</strong> to map cell types to Leiden clusters;
these will appear in <em>cell_metadata</em> within the
<strong>giottoObject</strong>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Name clusters</span></span>
<span><span class="va">clusters_cell_types_hypo</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Inhibitory'</span>, <span class="st">'Inhibitory'</span>, <span class="st">'Excitatory'</span>, <span class="st">'Astrocyte'</span>,<span class="st">'OD Mature'</span>, <span class="st">'Endothelial'</span>,</span>
<span>                             <span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span>,  <span class="st">'Ambiguous'</span>,<span class="st">'Ependymal'</span>, <span class="st">'Endothelial'</span>, <span class="st">'Microglia'</span>, <span class="st">'OD Mature'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">clusters_cell_types_hypo</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">cluster_order</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">merFISH_gobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, </span>
<span>                                 annotation_vector <span class="op">=</span> <span class="va">clusters_cell_types_hypo</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">'leiden_0.2_200'</span>, </span>
<span>                                 name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## show heatmap</span></span>
<span><span class="fu"><a href="../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>                    expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'cell_types'</span><span class="op">)</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">selected_genes</span>,</span>
<span>                    custom_feat_order <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/flip.html" class="external-link">rev</a></span><span class="op">(</span><span class="va">selected_genes</span><span class="op">)</span>,</span>
<span>                    custom_cluster_order <span class="op">=</span> <span class="va">clusters_cell_types_hypo</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/12-plotMetaDataHeatmap.png" style="width:50.0%"></p>
</div>
<div class="section level2">
<h2 id="part-8-visualize">Part 8: Visualize<a class="anchor" aria-label="anchor" href="#part-8-visualize"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Assign colors to each cell type </span></span>
<span><span class="va">mycolorcode</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'red'</span>, <span class="st">'lightblue'</span>, <span class="st">'yellowgreen'</span>,<span class="st">'purple'</span>, <span class="st">'darkred'</span>,</span>
<span>                <span class="st">'magenta'</span>, <span class="st">'mediumblue'</span>, <span class="st">'yellow'</span>, <span class="st">'gray'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mycolorcode</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Inhibitory'</span>, <span class="st">'Excitatory'</span>,<span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span>, </span>
<span>                       <span class="st">'Astrocyte'</span>, <span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>,<span class="st">'Endothelial'</span>, <span class="st">'Ambiguous'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, </span>
<span>            point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/13-UMAP_3D.png"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/14-spat3D.png"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/15-spatPlot2D.png" style="width:50.0%"></p>
<div class="section level3">
<h3 id="excitatory-cells-only">Excitatory Cells Only<a class="anchor" aria-label="anchor" href="#excitatory-cells-only"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">'Excitatory'</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/16-spat3D.png"></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">'Excitatory'</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/17-spatPlot2D.png" style="width:50.0%"></p>
</div>
<div class="section level3">
<h3 id="inhibitory-cells-only">Inhibitory Cells Only<a class="anchor" aria-label="anchor" href="#inhibitory-cells-only"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">'Inhibitory'</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/18-spat3D.png"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="st">'Inhibitory'</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/19-spatPlot2D.png" style="width:50.0%"></p>
</div>
<div class="section level3">
<h3 id="od-and-astrocytes-only">OD and Astrocytes Only<a class="anchor" aria-label="anchor" href="#od-and-astrocytes-only"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Astrocyte'</span>, <span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/20-spat3D.png"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Astrocyte'</span>, <span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/21-spatPlot2D.png" style="width:50.0%"></p>
</div>
<div class="section level3">
<h3 id="other-cells-only">Other Cells Only<a class="anchor" aria-label="anchor" href="#other-cells-only"></a>
</h3>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">merFISH_gobject</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>, <span class="st">'Endothelial'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/22-spat3D.png"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">merFISH_gobject</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>, <span class="st">'Endothelial'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/MerFISH_hypoth/220915_results/23-spatPlot2D.png" style="width:50.0%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com/" class="external-link">Ruben Dries</a>, Jiaji George Chen, Natalie Del Rossi, Emma Kelley, Junxiang Xu, Guo-Cheng Yuan.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
