<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Data Visualization • Giotto Suite</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Data Visualization">
<meta property="og:description" content="Giotto">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-BCXT5392QC"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-BCXT5392QC');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Giotto Suite</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0.997</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Start
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/getting_started.html#installation">Installation</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#getting-started">Getting Started</a>
    </li>
    <li>
      <a href="../articles/getting_started.html#analyses">Analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Datasets
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datasets_overview.html#array-based-datasets-1">Array based</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#single-cell-1">Single-cell</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#subcellular-datasets-1">Subcellular</a>
    </li>
    <li>
      <a href="../articles/datasets_overview.html#protein-multiplexing-1">Protein multiplexing</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/contribute.html">Contribute</a>
</li>
<li>
  <a href="../articles/giotto_news.html">News</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    FAQ
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/faqs.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../articles/github_issues.html">GitHub Issues Guidelines</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Try Giotto
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Docker</li>
    <li class="dropdown-header">Binder</li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RubD/Giotto/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Data Visualization</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RubD/Giotto/blob/HEAD/vignettes/Visualizations.Rmd" class="external-link"><code>vignettes/Visualizations.Rmd</code></a></small>
      <div class="hidden name"><code>Visualizations.Rmd</code></div>

    </div>

    
    
<p>This tutorial walks through the visualization capabilities of Giotto.
The clustering and dimension reduction methods focused on within the
previous tutorial, <a href="./dimension_reduction.html">dimension
reduction</a>, will be revisited and utilized to create heatmaps, violin
plots, and visualizations that are unique to Giotto: spatial maps and
networks.</p>
<p>This tutorial uses a merFISH dataset of mouse hypothalamic preoptic
regions from <a href="https://doi.org/10.1126/science.aau5324" class="external-link">Moffitt
et al.</a> A complete walkthrough of that dataset can be found <a href="./merFISH_hypoth_220915.html"><strong>here</strong></a>. To
download the data used to create the Giotto Object below, please ensure
that <a href="https://www.gnu.org/software/wget/?" class="external-link">wget</a> is installed
locally.</p>
<div class="section level2">
<h2 id="create-the-giotto-object">Create the Giotto Object<a class="anchor" aria-label="anchor" href="#create-the-giotto-object"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rubd.github.io/Giotto/" class="external-link">Giotto</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Specify path from which data may be retrieved/stored</span></span>
<span><span class="va">data_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_visual_data/'</span><span class="op">)</span></span>
<span><span class="co"># alternatively, "/path/to/where/the/data/lives/"</span></span>
<span></span>
<span><span class="co"># Specify path to which results may be saved</span></span>
<span><span class="va">results_directory</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/gobject_visual_results/'</span><span class="op">)</span> </span>
<span><span class="co"># alternatively, "/path/to/store/the/results/"</span></span>
<span></span>
<span><span class="co"># Ensure Giotto environment lives local to the machine</span></span>
<span><span class="va">genv_exists</span> <span class="op">=</span> <span class="fu"><a href="../reference/checkGiottoEnvironment.html">checkGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="va">genv_exists</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co"># The following command need only be run once to install the Giotto environment.</span></span>
<span>  <span class="fu"><a href="../reference/installGiottoEnvironment.html">installGiottoEnvironment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Optional: Specify a path to a Python executable within a conda or miniconda </span></span>
<span><span class="co"># environment. If set to NULL (default), the Python executable within the previously</span></span>
<span><span class="co"># installed Giotto environment will be used.</span></span>
<span><span class="va">my_python_path</span> <span class="op">=</span> <span class="cn">NULL</span> <span class="co"># alternatively, "/local/python/path/python" if desired.</span></span>
<span></span>
<span><span class="co"># Get the dataset</span></span>
<span><span class="fu"><a href="../reference/getSpatialDataset.html">getSpatialDataset</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">'merfish_preoptic'</span>, directory <span class="op">=</span> <span class="va">data_directory</span>, method <span class="op">=</span> <span class="st">'wget'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Giotto instructions and data preparation</span></span>
<span><span class="co"># Optional: Set Giotto instructions</span></span>
<span><span class="va">instrs</span> <span class="op">=</span> <span class="fu"><a href="../reference/createGiottoInstructions.html">createGiottoInstructions</a></span><span class="op">(</span>save_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  save_dir <span class="op">=</span> <span class="va">results_directory</span>, </span>
<span>                                  python_path <span class="op">=</span> <span class="va">my_python_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create file paths to feed data into Giotto object </span></span>
<span><span class="va">expr_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_data_expression.txt.gz"</span><span class="op">)</span></span>
<span><span class="va">loc_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_data_cell_locations.txt"</span><span class="op">)</span></span>
<span><span class="va">meta_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">data_directory</span>, <span class="st">"merFISH_3D_metadata.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Create Giotto object</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createGiottoObject.html">createGiottoObject</a></span><span class="op">(</span>expression <span class="op">=</span> <span class="va">expr_path</span>,</span>
<span>                              spatial_locs <span class="op">=</span> <span class="va">loc_path</span>,</span>
<span>                              instructions <span class="op">=</span> <span class="va">instrs</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Add additional metadata</span></span>
<span><span class="va">metadata</span> <span class="op">=</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">meta_path</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellMetadata.html">addCellMetadata</a></span><span class="op">(</span><span class="va">testobj</span>, </span>
<span>                          new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">layer_ID</span>, </span>
<span>                          vector_name <span class="op">=</span> <span class="st">'layer_ID'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/addCellMetadata.html">addCellMetadata</a></span><span class="op">(</span><span class="va">testobj</span>, </span>
<span>                          new_metadata <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">orig_cell_types</span>, </span>
<span>                          vector_name <span class="op">=</span> <span class="st">'orig_cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Process the Giotto Object</span></span>
<span><span class="co"># Note that for the purposes of this tutorial, the entire dataset will be visualized. </span></span>
<span><span class="co"># Thus, filter parameters are set to 0, so as to not remove any cells.</span></span>
<span><span class="co"># Note that since adjustment is not required, adjust_params is set to NULL.</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/processGiotto.html">processGiotto</a></span><span class="op">(</span><span class="va">testobj</span>,</span>
<span>                         filter_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                                              feat_det_in_min_cells <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                                              min_det_feats_per_cell <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>                         norm_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>norm_methods <span class="op">=</span> <span class="st">'standard'</span>, </span>
<span>                                            scale_feats <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                            scalefactor <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span>,</span>
<span>                         stat_params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>expression_values <span class="op">=</span> <span class="st">'normalized'</span><span class="op">)</span>,</span>
<span>                         adjust_params <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level2">
<h2 id="visualize-the-dataset">Visualize the Dataset<a class="anchor" aria-label="anchor" href="#visualize-the-dataset"></a>
</h2>
<p>This dataset includes eight sequential slices. As such it can be
visualized both in 2D and 3D.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 2D</span></span>
<span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/0-spatPlot2D.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 3D</span></span>
<span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/1-spat3D.png" style="width:150.0%"><br></p>
</div>
<div class="section level2">
<h2 id="create-and-visualize-clusters">Create and Visualize Clusters<a class="anchor" aria-label="anchor" href="#create-and-visualize-clusters"></a>
</h2>
<p>First, run a PCA on the data. For the purposes of this tutorial, no
highly variable genes will be identified or used in the reduction. The
data will simply undergo a dimension reduction through PCA. Then, run a
UMAP on the data for pre-clustering visualization. The UMAP may be
plotted in 2D and 3D.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run PCA</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runPCA.html">runPCA</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, feats_to_use <span class="op">=</span> <span class="cn">NULL</span>, scale_unit <span class="op">=</span> <span class="cn">FALSE</span>, center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># Run UMAP</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runUMAP.html">runUMAP</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, n_components <span class="op">=</span> <span class="fl">3</span>, n_threads <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="co"># Plot UMAP in 2D</span></span>
<span><span class="fu"><a href="../reference/plotUMAP_2D.html">plotUMAP_2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> </span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/2-UMAP_2D.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot UMAP 3D</span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> </span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/3-UMAP_3D.png"><br> Now, the data may be clustered. Create a nearest network, and then
create Leiden clusters. The clusters may be visualized in 2D or 3D, as
well as upon the UMAP and within the tissue.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a k Nearest Network for clustering</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createNearestNetwork.html">createNearestNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, dimensions_to_use <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, k <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Preform Leiden clustering</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/doLeidenCluster.html">doLeidenCluster</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                           resolution <span class="op">=</span> <span class="fl">0.25</span>, </span>
<span>                           n_iterations <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                           name <span class="op">=</span> <span class="st">'leiden_0.25.200'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the clusters upon the UMAP</span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, cell_color <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>            show_center_label <span class="op">=</span> <span class="cn">F</span>, save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'leiden_0.25.200_UMAP_3D'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/leiden_0.25.200_UMAP_3D.png"></p>
<p>Visualize Leiden clusters within the tissue by creating a Spatial
Plot, grouping by layer_ID.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>           point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, </span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, </span>
<span>           cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">260</span>, <span class="fl">160</span>, <span class="fl">60</span>, <span class="op">-</span><span class="fl">40</span>, <span class="op">-</span><span class="fl">140</span>, <span class="op">-</span><span class="fl">240</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/4-spatPlot2D.png" style="width:50.0%"></p>
<p>Visualize expression levels within the tissue by creating a Spatial
Plot, grouping by layer_ID, and specifying <em>cell_color</em> as the
number of features detected per cell.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot cell_color as a representation of the number of features/ cell ("nr_feats")</span></span>
<span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'nr_feats'</span>, color_as_factor <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">260</span>, <span class="fl">160</span>, <span class="fl">60</span>, <span class="op">-</span><span class="fl">40</span>, <span class="op">-</span><span class="fl">140</span>, <span class="op">-</span><span class="fl">240</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/5-spatPlot2D.png" style="width:50.0%"><br></p>
</div>
<div class="section level2">
<h2 id="compare-clusters">Compare Clusters<a class="anchor" aria-label="anchor" href="#compare-clusters"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot a heatmap of clusters</span></span>
<span><span class="fu"><a href="../reference/showClusterHeatmap.html">showClusterHeatmap</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, save_plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/6-showClusterHeatmap.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot a dendogram to explore cluster similarity</span></span>
<span><span class="fu"><a href="../reference/showClusterDendrogram.html">showClusterDendrogram</a></span><span class="op">(</span><span class="va">testobj</span>, h <span class="op">=</span> <span class="fl">0.5</span>, rotate <span class="op">=</span> <span class="cn">T</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/7-showClusterDendrogram.png" style="width:50.0%"><br></p>
</div>
<div class="section level2">
<h2 id="visualize-cell-markers">Visualize Cell Markers<a class="anchor" aria-label="anchor" href="#visualize-cell-markers"></a>
</h2>
<p>Marker features may be identified by calling
<strong>findMarkers_one_vs_all</strong>. This function detects
differentially expressed features by comparing a single cluster to all
others. Currently, three methods are supported: “<em>scran</em>”,
“<em>gini</em>”, and “<em>mast</em>”. Here, the “<em>gini</em>” method
is employed; details on the gini method may be found <a href="../docs/reference/findGiniMarkers.html">here</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="../reference/findMarkers_one_vs_all.html">findMarkers_one_vs_all</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>,</span>
<span>                                 method <span class="op">=</span> <span class="st">'gini'</span>,</span>
<span>                                 expression_values <span class="op">=</span> <span class="st">'normalized'</span>,</span>
<span>                                 cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>,</span>
<span>                                 min_feats <span class="op">=</span> <span class="fl">1</span>, rank_score <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Create a violinplot</span></span>
<span><span class="va">topgini_genes</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">markers</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">.SD</span>, <span class="fl">2</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">'cluster'</span><span class="op">]</span><span class="op">$</span><span class="va">feats</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/violinPlot.html">violinPlot</a></span><span class="op">(</span><span class="va">testobj</span>, feats <span class="op">=</span> <span class="va">topgini_genes</span>, cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, strip_position <span class="op">=</span> <span class="st">'right'</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/8-violinPlot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a heatmap of top gini genes by cluster</span></span>
<span><span class="fu"><a href="../reference/plotMetaDataHeatmap.html">plotMetaDataHeatmap</a></span><span class="op">(</span><span class="va">testobj</span>, expression_values <span class="op">=</span> <span class="st">'scaled'</span>,</span>
<span>                    metadata_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'leiden_0.25.200'</span><span class="op">)</span>,</span>
<span>                    selected_feats <span class="op">=</span> <span class="va">topgini_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/9-plotMetaDataHeatmap.png" style="width:50.0%"><br></p>
</div>
<div class="section level2">
<h2 id="visualize-cell-types-in-tissue">Visualize Cell Types in Tissue<a class="anchor" aria-label="anchor" href="#visualize-cell-types-in-tissue"></a>
</h2>
<p>To do this, the Leiden clusters must be annotated. Leveraging the
provided cell metadata and Giotto Spatial Plots, Leiden clusters may be
manually assigned a cell type. Alternative approaches (i.e. in the
absence of cell metadata with cell type identification ) could involve
the analysis of each cluster for enrichment in cell-specific marker
genes.</p>
<p>Since cell type annotations are included within the metadata that was
loaded into the Giotto Object, the UMAP may be plotted with cell-type
annotations. If cell types are known, Leiden clusters may be manually
assigned to a cell type, as will be done here.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the UMAP, annotated by cell type. </span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span><span class="va">testobj</span>, </span>
<span>            cell_color <span class="op">=</span> <span class="st">'orig_cell_types'</span>, </span>
<span>            save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'Original_Cell_Types_UMAP_3D'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/Original_Cell_Types_UMAP_3D.png"></p>
<p><br></p>
<p>Manually assign cell types to clusters via inspection of UMAP plots.
Specifically, the UMAP plots saved as <em>“leiden_0.25.200_UMAP3D”</em>
and <em>“Original_Cell_Types_UMAP3D”</em> are being compared for
assignment.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Manually assign Leiden clusters to a cell type</span></span>
<span><span class="va">cluster_range</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/terra/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">testobj</span><span class="op">@</span><span class="va">cell_metadata</span><span class="op">$</span><span class="va">cell</span><span class="op">$</span><span class="va">rna</span><span class="op">$</span><span class="va">leiden_0.25.200</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Note that cell types were condensed (i.e. "Endothelial 1", "Endothelial 2", ... were</span></span>
<span><span class="co"># combined into one cell type "Endothelial")</span></span>
<span><span class="va">manual_cluster</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Inhibitory'</span>, <span class="st">'Excitatory'</span>, <span class="st">'Inhibitory'</span>, <span class="st">'Astrocyte'</span>, <span class="st">'OD Mature'</span>, </span>
<span>                   <span class="st">'Endothelial'</span>, <span class="st">'Microglia'</span>, <span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span>, <span class="st">'Astrocyte'</span>,</span>
<span>                   <span class="st">'Ependymal'</span>, <span class="st">'Pericytes'</span>, <span class="st">'Ambiguous'</span>, <span class="st">'Microglia'</span>, <span class="st">'Inhibitory'</span>, <span class="st">'Inhibitory'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">manual_cluster</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/terra/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">cluster_range</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                         annotation_vector <span class="op">=</span> <span class="va">manual_cluster</span>,</span>
<span>                         cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, </span>
<span>                         name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_types_in_plot</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Inhibitory'</span>, <span class="st">'Excitatory'</span>,<span class="st">'OD Mature'</span>, <span class="st">'OD Immature'</span>, </span>
<span>                       <span class="st">'Astrocyte'</span>, <span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>,<span class="st">'Endothelial'</span>,</span>
<span>                       <span class="st">'Pericytes'</span>, <span class="st">'Ambiguous'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># This Giotto function will provide a distinct color palette. Colors </span></span>
<span><span class="co"># may change each time the function is run. </span></span>
<span><span class="va">mycolorcode</span> <span class="op">=</span> <span class="fu"><a href="../reference/getDistinctColors.html">getDistinctColors</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cell_types_in_plot</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mycolorcode</span><span class="op">)</span> <span class="op">=</span> <span class="va">cell_types_in_plot</span></span>
<span></span>
<span><span class="co"># Visualize the assigned types in the UMAP</span></span>
<span><span class="fu"><a href="../reference/plotUMAP_3D.html">plotUMAP_3D</a></span><span class="op">(</span><span class="va">testobj</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, point_size <span class="op">=</span> <span class="fl">1.5</span>, </span>
<span>            cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>            save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'manual_cluster_typing_UMAP_3D'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/manual_cluster_typing_UMAP_3D.png"></p>
<p>Now that each Leiden cluster has an associated cell type, cell types
may be viewed in tissue in 2D and in 3D within a Spatial Plot by
specifying the <em>cell_color</em> parameter as the name of the
annotation, ‘cell_types’.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span> <span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, </span>
<span>           cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/10-spatPlot2D.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">testobj</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/11-spat3D.png"></p>
<p><br></p>
<p>The plots may be subset by cell type in 2D and 3D.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot2D.html">spatPlot2D</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, point_size <span class="op">=</span> <span class="fl">1.0</span>, </span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>, <span class="st">'Endothelial'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span>,</span>
<span>           group_by <span class="op">=</span> <span class="st">'layer_ID'</span>, cow_n_col <span class="op">=</span> <span class="fl">2</span>, group_by_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">260</span>, <span class="op">-</span><span class="fl">290</span>, <span class="op">-</span><span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/12-spatPlot2D.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot3D.html">spatPlot3D</a></span><span class="op">(</span><span class="va">testobj</span>,</span>
<span>           cell_color <span class="op">=</span> <span class="st">'cell_types'</span>, axis_scale <span class="op">=</span> <span class="st">'real'</span>,</span>
<span>           sdimx <span class="op">=</span> <span class="st">'sdimx'</span>, sdimy <span class="op">=</span> <span class="st">'sdimy'</span>, sdimz <span class="op">=</span> <span class="st">'sdimz'</span>,</span>
<span>           show_grid <span class="op">=</span> <span class="cn">F</span>, cell_color_code <span class="op">=</span> <span class="va">mycolorcode</span>,</span>
<span>           select_cell_groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Microglia'</span>, <span class="st">'Ependymal'</span>, <span class="st">'Endothelial'</span><span class="op">)</span>, show_other_cells <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/13-spat3D.png"><br></p>
</div>
<div class="section level2">
<h2 id="visualize-cell-networks">Visualize Cell Networks<a class="anchor" aria-label="anchor" href="#visualize-cell-networks"></a>
</h2>
<p>It is preferred to use Delaunay geometry to create spatial networks.
In other cases, k-nearest neighbor may be used to create a spatial
network. Specifying the <em>method</em> parameter within <a href="../docs/reference/createSpatialNetwork.html"><strong>createSpatialNetwork</strong></a>
will accomplish this. By default, this function runs the Delaunay
method. Here, both methods, as well as potential modifications to the
k-nearest networks, will be shown.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Spatial Networks</span></span>
<span><span class="co"># The following fuction provides insight to the Delaunay Network. It has been </span></span>
<span><span class="co"># omitted graphically for the sake of this tutorial, but will be shown in-console</span></span>
<span><span class="co"># if this command is run.</span></span>
<span><span class="fu"><a href="../reference/plotStatDelaunayNetwork.html">plotStatDelaunayNetwork</a></span><span class="op">(</span>gobject<span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                        method <span class="op">=</span> <span class="st">'delaunayn_geometry'</span>, </span>
<span>                        maximum_distance <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                        show_plot <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                        save_plot <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create Spatial Network using Delaunay geometry</span></span>
<span><span class="va">testobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                               delaunay_method <span class="op">=</span> <span class="st">'delaunayn_geometry'</span>, </span>
<span>                               minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                               maximum_distance_delaunay <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create Spatial Networks using k-nearest neighbor with varying specifications</span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                                method <span class="op">=</span> <span class="st">'kNN'</span>, </span>
<span>                                k <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                name <span class="op">=</span> <span class="st">'spatial_network'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                                method <span class="op">=</span> <span class="st">'kNN'</span>, </span>
<span>                                k <span class="op">=</span> <span class="fl">10</span>, </span>
<span>                                name <span class="op">=</span> <span class="st">'large_network'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">testobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createSpatialNetwork.html">createSpatialNetwork</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">testobj</span>, </span>
<span>                                method <span class="op">=</span> <span class="st">'kNN'</span>, </span>
<span>                                k <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                                maximum_distance_knn <span class="op">=</span> <span class="fl">200</span>, </span>
<span>                                minimum_k <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                name <span class="op">=</span> <span class="st">'distance_network'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Now, visualize the different spatial networks in one layer of the dataset</span></span>
<span><span class="co"># Here layer 260 is selected, and only high expressing cells are included</span></span>
<span><span class="va">cell_metadata</span> <span class="op">=</span> <span class="fu"><a href="../reference/pDataDT.html">pDataDT</a></span><span class="op">(</span><span class="va">testobj</span><span class="op">)</span></span>
<span><span class="va">highexp_ids</span> <span class="op">=</span> <span class="va">cell_metadata</span><span class="op">[</span><span class="va">layer_ID</span><span class="op">==</span><span class="fl">260</span><span class="op">]</span><span class="op">[</span><span class="va">total_expr</span><span class="op">&gt;=</span><span class="fl">100</span><span class="op">]</span><span class="op">$</span><span class="va">cell_ID</span></span>
<span><span class="va">subtestobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/subsetGiotto.html">subsetGiotto</a></span><span class="op">(</span><span class="va">testobj</span>, cell_ids <span class="op">=</span> <span class="va">highexp_ids</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Re-annotate the subset Giotto Object</span></span>
<span><span class="va">subtestobj</span> <span class="op">=</span> <span class="fu"><a href="../reference/annotateGiotto.html">annotateGiotto</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subtestobj</span>, </span>
<span>                            annotation_vector <span class="op">=</span> <span class="va">manual_cluster</span>,</span>
<span>                            cluster_column <span class="op">=</span> <span class="st">'leiden_0.25.200'</span>, </span>
<span>                            name <span class="op">=</span> <span class="st">'cell_types'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subtestobj</span>, show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'Delaunay_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">1.5</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'Delaunay_network_spatPlot'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/Delaunay_network_spatPlot.png"></p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subtestobj</span>, show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'spatial_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'spatial_network_spatPlot'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/spatial_network_spatPlot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subtestobj</span>, show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'large_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'large_network_spatPlot'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/large_network_spatPlot.png" style="width:50.0%"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/spatPlot.html">spatPlot</a></span><span class="op">(</span>gobject <span class="op">=</span> <span class="va">subtestobj</span>, show_network <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         network_color <span class="op">=</span> <span class="st">'blue'</span>, spatial_network_name <span class="op">=</span> <span class="st">'distance_network'</span>,</span>
<span>         point_size <span class="op">=</span> <span class="fl">2.5</span>, cell_color <span class="op">=</span> <span class="st">'cell_types'</span>,</span>
<span>         save_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>save_name <span class="op">=</span> <span class="st">'distance_network_spatPlot'</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../inst/images/getting_started_figs/visualizations/distance_network_spatPlot.png" style="width:50.0%"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.drieslab.com/" class="external-link">Ruben Dries</a>, Jiaji George Chen, Natalie Del Rossi, Emma Kelley, Junxiang Xu, Guo-Cheng Yuan.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
